packages/x-date-pickers/src/LocalizationProvider/LocalizationProvider.tsx

```diff
@@ -28,7 +28,10 @@ if (process.env.NODE_ENV !== 'production') {
 
 export interface LocalizationProviderProps<TDate> {
   children?: React.ReactNode;
-  /** DateIO adapter class function */
+  /**
+   * Date library adapter class function.
+   * @see See the {@link https://next.mui.com/x/react-date-pickers/adapters-locale/#set-a-custom-locale available adapters page} for more details.
```

Comment:
The page you are pointing to explains how to use `adapterLocale`, not the `dateAdapter`
```suggestion
   * @see See the {@link https://next.mui.com/x/react-date-pickers/#date-library available adapters page} for more details.
```

Reply:
I was thinking more about the examples of adapters usage and that page you linked has no examples...
Maybe the way could be to add a section referencing my mentioned page in your suggested one to sort of link them together. 🤔

Reply:
For a docs example, you have the following link which I an interactive demo: https://next.mui.com/x/react-date-pickers/getting-started/#code-setup

The problem with the current link is that you land on a screen where everything is dedicated to dayjs, so my first idea was

> Ok I will not find the information if I want to use luxon, or another adapter

![image](https://user-images.githubusercontent.com/45398769/218460663-b82393cc-76e4-493b-98b6-ccfa398c9530.png)

Reply:
> The problem with the current link is that you land on a screen where everything is dedicated to dayjs, so my first idea was
> 
> > Ok I will not find the information if I want to use luxon, or another adapter

Fair point, it's not the best page to link to in order to have the best initial visibility, but actually so are other options. 🙈 By the way, the sidebar links do have all the adapter libraries listed. 🙈 



> For a docs example, you have the following link which I an interactive demo: https://next.mui.com/x/react-date-pickers/getting-started/#code-setup

I actually think that this is the best link we can have for `dateAdapter` doc, wouldn't you agree? 🤔